compiler/rustc_resolve/src/imports.rs

```diff
@@ -554,6 +555,40 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
     }
 
+    pub(crate) fn check_local_binding_shadows_glob_reexport(&mut self) {
+        for module in self.arenas.local_modules().iter() {
+            let resolutions = self.resolutions(module).borrow();
+            for (_, resolution) in resolutions.iter() {
+                let resolution = resolution.borrow();
+                if let Some(binding) = resolution.binding
+                && let Some(glob_binding) = resolution.shadowed_glob
+                && let NameBinding { kind: NameBindingKind::Import { import, .. }, .. } = glob_binding
+                && let Some(import_id) = import.id()
+                && let import_local_def_id = self.local_def_id(import_id)
```

Comment:
```suggestion
                && let import_def_id = self.local_def_id(import_id)
```
Nit: this is typically not said explicitly unless there's some ambiguity between local and non-local ids.